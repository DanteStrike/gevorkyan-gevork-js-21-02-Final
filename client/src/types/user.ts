export type UserTitleType = 'mr' | 'ms' | 'mrs' | 'miss' | 'dr' | '';
export type UserGenderType = 'male' | 'female' | 'other' | '';

export interface IUserPreview {
  id: string;
  title?: UserTitleType;
  firstName: string;
  lastName: string;
  picture?: string;
}

export interface ILocation {
  street: string;
  city: string;
  state: string;
  country: string;
  timezone: string;
}

export interface IUser extends IUserPreview {
  gender: UserGenderType;
  email: string;
  dateOfBirth: string;
  registerDate: string;
  phone: string;
  location: ILocation;
}

export interface IUserRegistration {
  firstName: string;
  lastName: string;
  gender: UserGenderType;
  dateOfBirth: string;
  email: string;
  phone: string;
}

export interface IUserUpdate extends Omit<IUserRegistration, 'email'> {
  id: string;
  picture: string;
}
