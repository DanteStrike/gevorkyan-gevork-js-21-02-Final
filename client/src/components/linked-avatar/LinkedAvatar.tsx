import React from 'react';
import {Avatar} from 'antd';
import CustomLink from '../custom-link/CustomLink';
import {IUserPreview} from '../../types';

function LinkedAvatar({user, className, to}: {user: IUserPreview; className: string; to: string}) {
  return (
    <CustomLink to={to} className={className}>
      <Avatar src={user.picture} />
    </CustomLink>
  );
}

export default React.memo(LinkedAvatar);
