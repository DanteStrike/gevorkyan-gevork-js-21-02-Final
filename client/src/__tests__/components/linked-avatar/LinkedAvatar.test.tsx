import {shallow} from 'enzyme';
import React from 'react';
import LinkedAvatar from '../../../components/linked-avatar/LinkedAvatar';
import {userTwo} from '../../mocks/user.mock';

describe(`Component LinkedAvatar should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<LinkedAvatar user={userTwo} className="anyClass" to="anyRoute" />);
    expect(wrapper).toMatchSnapshot();
  });
});
