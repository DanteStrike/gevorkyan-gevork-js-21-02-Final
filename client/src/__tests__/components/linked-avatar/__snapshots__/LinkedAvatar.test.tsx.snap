// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LinkedAvatar should render correctly snapshot default test 1`] = `
<Memo(CustomLink)
  className="anyClass"
  to="anyRoute"
>
  <Avatar
    shape="circle"
    size="default"
    src="url"
  />
</Memo(CustomLink)>
`;
